src/app/ClusterInfo.h

```diff
@@ -25,96 +25,54 @@
 
 namespace chip {
 namespace app {
-static constexpr AttributeId kRootAttributeId = 0xFFFFFFFF;
 
+/**
+ * ClusterInfo is the internal representation of an attribute path or an event path. It uses some invalid values for representing
+ * the missing of one or more fields (and wildcard paths according to the context.)
+ */
 struct ClusterInfo
 {
-    enum class Flags : uint8_t
-    {
-        kFieldIdValid   = 0x01,
-        kListIndexValid = 0x02,
-        kEventIdValid   = 0x03,
-    };
+    // 0x0000'0000'0000'0000 is unspecified node id in Matter.
+    static constexpr NodeId kInvalidNodeId = 0x0000'0000'0000'0000;
+    // Endpoint Id is a uint16 number, and should between 0 and 0xFFFE
+    static constexpr EndpointId kInvalidEndpointId = 0xFFFF;
+    // The ClusterId, AttributeId and EventId are MEIs,
+    // 0xFFFF is not a valid manufactor code, thus 0xFFFF'FFFF is not a valid MEI
+    static constexpr ClusterId kInvalidClusterId     = 0xFFFF'FFFF;
```

Comment:
Would prefer `kInvalidClusterId`, since the wildcard id is not part of the spec, and invalid is a better name which indicates the field should not be put into the tlv when encoding.

Also provided some functions so the `IsAttributePathSupersetOf` looks better.

Reply:
How about `kWildcardClusterValue`?  The fact that we use "invalid" to indicate "wildcard" is pretty non-obvious.  But maybe if it's hidden behind accessors it's not too bad....

Reply:
If we could re-name the accessors to indicate the wildcard-ness, I'd be fine.

Reply:
Updated accessors to HasWildcard*()